lighthouse-core/gather/gatherers/seo/font-size.js

```diff
@@ -109,142 +185,175 @@ function getNodeTextLength(node) {
+      nodeId: node.parentId,
     });
+
+    const fontSizeProperty = computedStyle.find(({name}) => name === FONT_SIZE_PROPERTY_NAME);
+
+    return {
+      // @ts-ignore - font size property guaranteed to be returned in getComputedStyle
+      fontSize: parseInt(fontSizeProperty.value, 10),
+      textLength: getNodeTextLength(node),
+      node: /** @type {LH.Artifacts.FontSize.DomNodeWithParent} */ (node.parentNode),
+    };
+  } catch (err) {
+    Sentry.captureException(err, {tags: {gatherer: 'FontSize'}});
+    return null;
+  }
 }
 
 /**
  * @param {LH.Artifacts.FontSize.DomNodeWithParent} node
  * @returns {boolean}
  */
 function isNonEmptyTextNode(node) {
-  return node.nodeType === global.Node.TEXT_NODE &&
+  return (
+    node.nodeType === TEXT_NODE_TYPE &&
     !TEXT_NODE_BLOCK_LIST.has(node.parentNode.nodeName) &&
-    getNodeTextLength(node) > 0;
+    getNodeTextLength(node) > 0
+  );
 }
 
 class FontSize extends Gatherer {
+  /**
+   * @param {LH.Gatherer.PassContext['driver']} driver
+   */
+  static async fetchNodesToAnalyze(driver) {
+    let failingTextLength = 0;
+    let visitedTextLength = 0;
+    let totalTextLength = 0;
+
+    const nodes = await getAllNodesFromBody(driver);
+
+    const textNodes = nodes.filter(isNonEmptyTextNode);
+    totalTextLength = textNodes.reduce((sum, node) => (sum += getNodeTextLength(node)), 0);
+
+    const nodesToVisit = textNodes
+      .sort((a, b) => getNodeTextLength(b) - getNodeTextLength(a))
+      .slice(0, MAX_NODES_VISITED);
+
+    const nodePromises = nodesToVisit.map(node => fetchComputedFontSize(driver, node));
+    const visitedNodes = await Promise.all(nodePromises);
+
+    /** @type {Array<NodeFontData>} */
+    const failingNodes = [];
+    for (const visitedNode of visitedNodes) {
+      if (!visitedNode) continue;
+      visitedTextLength += visitedNode.textLength;
+
+      if (visitedNode.fontSize < MINIMAL_LEGIBLE_FONT_SIZE_PX) {
+        failingNodes.push(visitedNode);
+        failingTextLength += visitedNode.textLength;
+      }
+    }
+
+    const nodesToAnalyze = failingNodes
+      .sort((a, b) => b.textLength - a.textLength)
+      .slice(0, MAX_NODES_ANALYZED);
+
+    return {totalTextLength, visitedTextLength, failingTextLength, nodesToAnalyze};
+  }
+
+  /**
+   *
+   * @param {LH.Gatherer.PassContext['driver']} driver
+   * @param {Array<NodeFontData>} nodesToAnalyze
+   */
+  static async analyzeFailingNodes(driver, nodesToAnalyze) {
+    const analysisPromises = nodesToAnalyze.map(async failingNode => {
+      failingNode.cssRule = await fetchSourceRule(driver, failingNode.node);
+      return failingNode;
+    });
+
+    const analyzedFailingNodesData = await Promise.all(analysisPromises);
+
+    const analyzedFailingTextLength = analyzedFailingNodesData.reduce(
+      (sum, {textLength}) => (sum += textLength),
+      0
+    );
+
+    return {analyzedFailingNodesData, analyzedFailingTextLength};
```

Comment:
just `analyzedFailingNodes`, maybe?

Reply:
well that's just based on the artifact shape name, feel strongly enough to change the artifact type and its usages up the chain?

Reply:
> well that's just based on the artifact shape name, feel strongly enough to change the artifact type and its usages up the chain?

ha, I didn't before, but I think we maybe should. The `Data` part is definitely redundant, but what do you think about `sourcedFailingNodes`?